Src/Support/Google.Apis.Auth/OAuth2/ComputeCredential.cs

```diff
@@ -94,6 +96,38 @@ public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskC
 
         #endregion
 
+        /// <inheritdoc/>
+        public Task<OIdCToken> GetOIdCTokenAsync(OIdCTokenOptions options, CancellationToken cancellationToken = default)
+        {
+            options.ThrowIfNull(nameof(options));
+            // If at some point some properties are added to OIdCToken that depend on the token having been fetched
+            // then initialize the token here.
+            TokenRefreshManager tokenRefreshManager = null;
+            tokenRefreshManager = new TokenRefreshManager(
+                ct => RefreshOIdCTokenAsync(tokenRefreshManager, options, ct), Clock, Logger);
+            return Task.FromResult(new OIdCToken(tokenRefreshManager));
+        }
+
+        private async Task<bool> RefreshOIdCTokenAsync(TokenRefreshManager caller, OIdCTokenOptions options, CancellationToken cancellationToken)
+        {
+            string uri = $"{GoogleAuthConsts.ComputeOidcTokenUrl}?audience={options.TargetAudience}";
+            if (options.TokenFormat == OIdCTokenFormat.Full || options.TokenFormat == OIdCTokenFormat.FullWithLicences)
+            {
+                uri = $"{uri}&format=FULL";
+                if (options.TokenFormat == OIdCTokenFormat.FullWithLicences)
+                {
+                    uri = $"{uri}&licenses=TRUE";
+                }
+            }
+
+            var httpRequest = new HttpRequestMessage(HttpMethod.Get, uri);
+            httpRequest.Headers.Add(MetadataFlavor, GoogleMetadataHeader);
+            var response = await HttpClient.SendAsync(httpRequest, cancellationToken).ConfigureAwait(false);
+            var token = await TokenResponse.FromHttpResponseAsync(response, Clock, Logger).ConfigureAwait(false);
```

Comment:
Does this do the appropriate "ensure success codes" etc?

Reply:
I'll check, here and in the existing `RequestAccessTokenAsync`.